src/Lunr/Ticks/PHPMailer/PHPMailer.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/**
+ * This file contains the PHPMailer class.
+ *
+ * SPDX-FileCopyrightText: Copyright 2025 Framna Netherlands B.V., Zwolle, The Netherlands
+ * SPDX-License-Identifier: MIT
+ */
+
+namespace Lunr\Ticks\PHPMailer;
+
+use Lunr\Ticks\AnalyticsDetailLevel;
+use Lunr\Ticks\EventLogging\EventLoggerInterface;
+use Lunr\Ticks\TracingControllerInterface;
+use Lunr\Ticks\TracingInfoInterface;
+use PHPMailer\PHPMailer\PHPMailer as BaseMailer;
+
+/**
+ * PHPMailer class
+ */
+class PHPMailer extends BaseMailer
+{
+
+    /**
+     * Shared instance of the event logger
+     * @var EventLoggerInterface
+     */
+    protected readonly EventLoggerInterface $eventLogger;
+
+    /**
+     * Shared instance of the info tracing controller
+     * @var TracingControllerInterface&TracingInfoInterface
+     */
+    protected readonly TracingControllerInterface&TracingInfoInterface $tracingController;
+
+    /**
+     * Profiling level
+     * @var AnalyticsDetailLevel
+     */
+    protected AnalyticsDetailLevel $analyticsDetailLevel;
+
+    /**
+     * Constructor.
+     *
+     * @param bool|null $exceptions Should we throw external exceptions?
+     */
+    public function __construct(?bool $exceptions = NULL)
+    {
+        $this->analyticsDetailLevel = AnalyticsDetailLevel::None;
+
+        parent::__construct($exceptions);
+    }
+
+    /**
+     * Destructor.
+     */
+    public function __destruct()
+    {
+        unset($this->analyticsDetailLevel);
+
+        parent::__destruct();
+    }
+
+    /**
+     * Enable PHPMailer request analytics.
+     *
+     * @param EventLoggerInterface                            $eventLogger Instance of an event logger
+     * @param TracingControllerInterface&TracingInfoInterface $controller  Instance of a tracing controller
+     * @param AnalyticsDetailLevel                            $level       Analytics detail level (defaults to Info)
+     *
+     * @return void
+     */
+    public function enableAnalytics(
+        EventLoggerInterface $eventLogger,
+        TracingControllerInterface&TracingInfoInterface $controller,
+        AnalyticsDetailLevel $level = AnalyticsDetailLevel::Info,
+    ): void
+    {
```

Comment:
In this case, I would pass the objects in the constructor.

The primary use-case of this particular class is for analytics. There's no other reason for using it, so it makes little sense to make analytics optional. The only classes where we do that are ones that have primarily a different focus. Like `LunrSoapClient` being primarily to enable dependency injection, or `MySQLConnection` having lots of other functionality before analytics.

It does make sense to have a separate setter method to override the analytics detail level though.